chef_postgres/libraries/pg_config.rb

```diff
@@ -155,25 +154,13 @@ def maintenance_work_memory
       def wal_keep_segments
         # Convert to modifier % of available disk space on log drive in MB, then divide by 16MB
         # If log location becomes configurable, the drive needs to also become configurable
-
         modifier = { web: 0.50,
                      oltp: 0.50,
                      dw: 0.50,
                      mixed: 0.50,
                      desktop: 0.25,
         }.fetch(workload)
-
-        (kb_available.to_f * modifier / 1024 / 16).round
-      end
-
-      def kb_available
-        return node["filesystem"][mount_drive]["kb_available"] unless node["filesystem"][mount_drive].nil?
-
-        kb = 0
-        node["filesystem"].each do |key, value|
-          kb = value['kb_available'].to_i if value['kb_available'].to_i > kb
-        end
-        kb
+          (node["filesystem"]["by_device"]["#{rootdrive}"]["kb_available"].to_f * modifier / 1024 / 16).round
```

Comment:
technically, you don't need to string interpolate a string.  So instead of `"#{rootdrive}"`, you can just have `rootdrive`

Reply:
updated